### DISCUSSION TITLE
NextAuth / AuthJs with i18n

### DISCUSSION THREAD

**Discussion by laurent512** [ivanhofer/typesafe-i18n#698] [General]
Hello, 

I have cloned your implementation with Sveltekit and am trying to integrate authentication with it. 
https://github.com/ivanhofer/typesafe-i18n-demo-sveltekit/tree/main

Would you have an idea to integrate it with NextAuth / AuthJS ? 
https://authjs.dev/reference/sveltekit

I am a bit confused on the hooks.server.ts part as both of the libraries have requirements on it :) 

Thank you so much for your help

**Comment by ivanhofer** [comment]
You could try it with the `sequence` helper function.

```ts
import { sequence} from "@sveltejs/kit";

export const handle = sequence(
	async ({ event, resolve }) => {
		// ... i18n `handle` code
	},
	SvelteKitAuth({
  		providers: [GitHub({ clientId: GITHUB_ID, clientSecret: GITHUB_SECRET })],
	}
);
```

You should open an issue on the `Auth.js` repo. They should add that to their docs as it will be often the case that someone needs custom `handle` funcitonality.

**Comment by laurent512** [comment]
Thank you so much , it worked!  :)
